src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI.Task/GenAPITask.cs

```diff
@@ -62,17 +65,25 @@ public class GenAPITask : TaskBase
         /// <inheritdoc />
         protected override void ExecuteCore()
         {
-            GenAPIApp.Run(new MSBuildLog(Log),
-                Assemblies!,
-                AssemblyReferences,
-                OutputPath,
-                HeaderFile,
-                ExceptionMessage,
-                ExcludeApiFiles,
-                ExcludeAttributesFiles,
-                RespectInternals,
-                IncludeAssemblyAttributes
-            );
+            Debug.Assert(Assemblies != null, "Assemblies cannot be null.");
+
+            ILog log = new MSBuildLog(Log);
+            (IAssemblySymbolLoader loader, Dictionary<string, IAssemblySymbol> assemblySymbols) = AssemblySymbolLoader.CreateFromFiles(
+                log,
+                assembliesPaths: Assemblies,
+                assemblyReferencesPaths: AssemblyReferences,
+                RespectInternals);
```

Comment:
The .Task and .Tool projects are the UI layer of GenAPI. The idea is that they only pass POCOs through and not instantiate heavy code like the assembly symbol loader. That's what the business layer is for: the GenAPI.csproj.

Reply:
I can create a helper class in GenApiApp that takes care of the instantiation of AssemblySymbolLoader.

Reply:
Done. I might be able to delete the other Run method I added in the next PR if it's not used.